common/v2/features/SendAssets/types.ts

```diff
@@ -19,50 +20,70 @@ export interface ITxObject {
 }
 
 export interface ITxConfig {
-  readonly gasLimit: string; // Move to BN
-  readonly gasPrice: string; // Move to BN
+  readonly gasLimit: string;
+  readonly gasPrice: string;
   readonly nonce: string;
-  readonly amount: string; // Move to BN
+  readonly amount: string;
+  readonly value: string;
+  readonly to: string;
+  readonly chainId: number;
   readonly data: string;
-  readonly receiverAddress: string; // Can't be an ExtendedAddressBook since recipient may not be registered
+  readonly receiverAddress: string;
   readonly senderAccount: IExtendedAccount;
-  readonly asset: IAsset | Asset;
-  readonly network?: Network;
+  readonly asset: Asset;
+  readonly network: Network;
 }
 
 export interface ITxReceipt {
   [index: string]: any;
 }
 
+export interface ITxData {
+  readonly hash: string;
+  readonly network: Network;
+  readonly asset: Asset | undefined;
+
+  readonly amount: string;
+  readonly value: string;
+  readonly to: string;
+  readonly from: string;
+  readonly nonce: number;
+
+  readonly gasLimit: string; // Hex
+  readonly gasPrice: string; // Hex - wei
+  readonly data: string; // Hex
+}
+
 export interface IFormikFields {
-  asset: IAsset | Asset;
+  asset: Asset;
   receiverAddress: string;
   amount: string;
   account: IExtendedAccount;
   txDataField: string;
   gasEstimates: GasEstimates;
-  gasPriceField: string; // Use only if advanced tab is open AND user has input gas price
+  gasPriceField: string;
   gasPriceSlider: string;
-  gasLimitField: string; // Use only if advanced tab is open AND isGasLimitManual is true
-  gasLimitEstimated: string;
+  gasLimitField: string;
   nonceField: string; // Use only if user has input a manual nonce value.
-  nonceEstimated: string;
   network: Network;
+  advancedTransaction: boolean;
   resolvedENSAddress: string; // Address returned when attempting to resolve an ENS/RNS address.
```

Comment:
We need to store and display the ens name in the field still, got another idea on how to keep that value before we overwrite it with the ENS name's resolved value?

Reply:
May be better to set a new prop for the ens name string (or address book string) and ensure the receiver address is always the most up to date address information that's been validated?

Maybe think of it like this:

- User input (we don't change this variable, just use it as input. The field in the UI contains whatever the user typed but doesn't ever go directly to transaction data.)

- Depending on what that input is, set new variables to receiver address, ens name + receiver address, address book label + receiver address, etc.

- control the UI based on what information is available. If receiver + user input match, don't show anything. If the ens name variable exists, show it. If the receiver + user input don't match, show the receiver address var, Etc.

- Always use receiver address in transaction and confirmation screens, regardless of where that information came from.

This would also likely make it easier to show relevant information the other way around, as we control the UI based on what information is available in the dedicated variables. E.g. showing an ens name or address book label when a user enters an address.

I do worry that the receiver address variable could get out of sync with the UI and/or we may miss bugs that only occur if user is using an ens name vs address.

Reply:
We're going to worry about this one in the ENS PR after this PR is merged c: